src/openfe_analysis/tests/test_reader.py

```diff
@@ -0,0 +1,176 @@
+             [0.95652354, 2.99566, 1.3466661],
+             [4.0027137, 4.695961, 3.6892936]]
+        ) * 10,
+    )
+    assert_allclose(
+        u.dimensions,
+        [82.12723, 82.12723, 82.12723, 90., 90., 90.]
+    )
+
+    # Finally we rewind to the second frame to make sure that's possible
+    u.trajectory[1]
+    assert u.trajectory.time == pytest.approx(1.0)
+    assert_allclose(
+        u.atoms[4:7].positions,
+        np.array([[6.78754, -1.2783755, 8.433636],
+                  [6.62524, -1.333609, 8.399696],
+                  [6.744502, -1.5663723, 8.332421]]) * 10,
+    )
+    assert_allclose(
+        u.dimensions,
+        [82.191055, 82.191055, 82.191055, 90., 90., 90.]
+    )
+
+
+def test_universe_from_nc_file(simulation_nc, hybrid_system_pdb):
+    ds = nc.Dataset(simulation_nc)
+
+    u = mda.Universe(hybrid_system_pdb, ds,
+                     format='MultiStateReporter', state_id=0)
+
+    assert u
+    assert len(u.atoms) == 4782
+    assert len(u.trajectory) == 501
+    assert u.trajectory.dt == pytest.approx(1.0)
+
+
+def test_universe_creation_noconversion(simulation_nc, hybrid_system_pdb):
+    u = mda.Universe(hybrid_system_pdb, simulation_nc,
+                     format=FEReader, state_id=0, convert_units=False)
+
+    assert_allclose(
+        u.atoms[:3].positions,
+        np.array([[6.51474, -1.7640617, 8.406607],
+                  [6.641961, -1.8410535, 8.433087],
+                  [6.71369, -1.8112476, 8.533738]]),
+    )
+
+
+def test_fereader_negative_state(simulation_nc, hybrid_system_pdb):
+    u = mda.Universe(
+        hybrid_system_pdb, simulation_nc, format=FEReader,
+        state_id=-1
+    )
+
+    assert u.trajectory._state_id == 10
+    assert u.trajectory._replica_id is None
+
+
+def test_fereader_negative_replica(simulation_nc, hybrid_system_pdb):
+    u = mda.Universe(
+        hybrid_system_pdb, simulation_nc, format=FEReader,
+        replica_id=-2
+    )
+
+    assert u.trajectory._state_id is None
+    assert u.trajectory._replica_id == 9
+
+
+@pytest.mark.parametrize('rep_id, state_id', [
+    [None, None],
+    [1, 1]
+])
+@pytest.mark.flaky(reruns=3)
+def test_fereader_replica_state_id_error(
+    simulation_nc, hybrid_system_pdb, rep_id, state_id
+):
+    with pytest.raises(ValueError, match="Specify one and only one"):
+        _ = mda.Universe(
+            hybrid_system_pdb, simulation_nc, format=FEReader,
+            state_id=state_id, replica_id=rep_id
+        )
+
+
+def test_simulation_skipped_nc(
+    simulation_skipped_nc, hybrid_system_skipped_pdb
+):
+    u = mda.Universe(
+        hybrid_system_skipped_pdb, simulation_skipped_nc,
+        format=FEReader, replica_id=0,
+    )
+
+    assert len(u.trajectory) == 6
+    assert u.trajectory.n_frames == 6
+    for ts in u.trajectory:
+        assert np.all(u.atoms.positions > 0)
+        with pytest.raises(mda.exceptions.NoDataError, match='This Timestep has no velocities'):
```

Comment:
Possibly do this outside of the loop? As far as I remember, MDAnallysis doesn't support variable velocities, so if it fails on one timestep it'll fail on all timesteps.